Sources/CardPayments/CardClient.swift

```diff
@@ -12,22 +12,84 @@ public class CardClient: NSObject {
     private let config: CoreConfig
     private let webAuthenticationSession: WebAuthenticationSession
     private var analyticsService: AnalyticsService?
+    private var graphQLClient: GraphQLClient?
 
     /// Initialize a CardClient to process card payment
     /// - Parameter config: The CoreConfig object
     public init(config: CoreConfig) {
         self.config = config
         self.apiClient = APIClient(coreConfig: config)
         self.webAuthenticationSession = WebAuthenticationSession()
+        self.graphQLClient = GraphQLClient(environment: config.environment)
     }
 
     /// For internal use for testing/mocking purpose
-    init(config: CoreConfig, apiClient: APIClient, webAuthenticationSession: WebAuthenticationSession) {
+    init(
+        config: CoreConfig,
+        apiClient: APIClient,
+        webAuthenticationSession: WebAuthenticationSession,
+        graphQLClient: GraphQLClient? = nil
+    ) {
         self.config = config
         self.apiClient = apiClient
         self.webAuthenticationSession = webAuthenticationSession
+        self.graphQLClient = graphQLClient
     }
-
+    
+    public func vault(vaultRequest: CardVaultRequest) {
+        Task {
+            guard let graphQLClient else {
+                throw CardClientError.unknownError
+            }
+            do {
+                let card = vaultRequest.card
+                let setupToken = vaultRequest.setupToken
+                let (updateResult) = try await updateSetupToken(
+                    vaultSetupToken: setupToken,
+                    card: card)
```

Comment:
```suggestion
                let (updateResult) = try await updateSetupToken(vaultSetupToken: setupToken, card: card)
```

Reply:
I got lint complaint about line being longer than 140 characters or something

Reply:
That was probably for a different line, this one should be fine as 1 line

Reply:
addressed in 8a67bce871356c3a8bd910be35cc7efd64d1864c